### ISSUE TITLE
Issue when reseting state when handling a return statement, Safari specific

### ISSUE THREAD

**Issue by psachs21** [facebook/draft-js#396]
So, weird issue specific to Safari. I reproduced it in the plaintext example by adding a return handler to the component.

```js
this.onReturn = () => {
  this.setState({editorState: EditorState.createEmpty()});
  return true;
}
...
<Editor
  ...
  handleReturn={this.onReturn}
  />
```
On the iOS simulator, if you type, and press enter, and then start typing again, the cursor and text content get quite confused. 

**Update** This is not specific to mobile safari, it applies also to desktop Safari

**Comment by psachs21** [comment]
Additional info: There are null exceptions in the console on safari, but I noticed that chrome also has a slightly strange behaviour in that after you clear the state, the next few keystrokes don't register properly.

**Comment by psachs21** [comment]
Turns out this was related to how i was clearing the state of the editor. Fixed in #445
